apps/api/src/campaign-application/campaign-application.service.spec.ts

```diff
@@ -20,131 +41,117 @@ describe('CampaignApplicationService', () => {
+        toEntity: new CreateCampaignApplicationDto().toEntity,
       }
 
-      expect(() => service.create(dto)).toThrow(
+      await expect(service.create(dto, mockPerson)).rejects.toThrow(
         new BadRequestException('All agreements must be checked'),
       )
     })
 
-    it('should throw an error if transparencyTermsAccepted is false', () => {
+    it('should throw an error if transparencyTermsAccepted is false', async () => {
       const dto: CreateCampaignApplicationDto = {
-        ...baseDto,
+        ...mockNewCampaignApplication,
         acceptTermsAndConditions: true,
         transparencyTermsAccepted: false,
         personalInformationProcessingAccepted: true,
+        toEntity: new CreateCampaignApplicationDto().toEntity,
       }
 
-      expect(() => service.create(dto)).toThrow(
+      await expect(service.create(dto, mockPerson)).rejects.toThrow(
         new BadRequestException('All agreements must be checked'),
       )
     })
 
-    it('should throw an error if personalInformationProcessingAccepted is false', () => {
+    it('should throw an error if personalInformationProcessingAccepted is false', async () => {
       const dto: CreateCampaignApplicationDto = {
-        ...baseDto,
+        ...mockNewCampaignApplication,
         acceptTermsAndConditions: true,
         transparencyTermsAccepted: true,
         personalInformationProcessingAccepted: false,
+        toEntity: new CreateCampaignApplicationDto().toEntity,
       }
 
-      expect(() => service.create(dto)).toThrow(
+      await expect(service.create(dto, mockPerson)).rejects.toThrow(
         new BadRequestException('All agreements must be checked'),
       )
     })
 
-    it('should add a new campaign-application if all agreements are true', () => {
+    it('should add a new campaign-application if all agreements are true', async () => {
       const dto: CreateCampaignApplicationDto = {
-        ...baseDto,
+        ...mockNewCampaignApplication,
         acceptTermsAndConditions: true,
         transparencyTermsAccepted: true,
         personalInformationProcessingAccepted: true,
+        toEntity: new CreateCampaignApplicationDto().toEntity,
       }
 
-      expect(service.create(dto)).toBe('This action adds a new campaignApplication')
+      const mockOrganizerId = 'mockOrganizerId'
+      jest.spyOn(mockOrganizerService, 'create').mockResolvedValue({
+        id: mockOrganizerId,
+        personId: mockPerson.id,
+      })
+
+      jest
+        .spyOn(prismaMock.campaignApplication, 'create')
+        .mockResolvedValue(mockCreatedCampaignApplication)
+
+      const result = await service.create(dto, mockPerson)
+
+      expect(result).toEqual(mockCreatedCampaignApplication)
+
+      expect(mockOrganizerService.create).toHaveBeenCalledWith({
+        personId: mockPerson.id,
+      })
+
+      expect(prismaMock.campaignApplication.create).toHaveBeenCalledWith({
+        data: {
+          campaignName: dto.campaignName.trim(),
```

Comment:
This looks like the logic itself and while it's technically correct I belive we'll get more value having hardcoded value here i.e.  {

data :{
campaignName: 'my-campaign',
organizerName: 'test organizer',

...